src/open-feature.ts

```diff
@@ -78,6 +91,23 @@ class OpenFeatureAPI implements GlobalApi {
   getContext(): EvaluationContext {
     return this._context;
   }
+
+  setTransactionContextPropagator(transactionContextPropagator: TransactionContextPropagator): OpenFeatureAPI {
+    this._transactionContextPropagator = transactionContextPropagator;
+    return this;
+  }
+
+  setTransactionContext<R>(
+    transactionContext: TransactionContext,
+    callback: (...args: unknown[]) => R,
+    ...args: unknown[]
+  ): void {
+    this._transactionContextPropagator.setTransactionContext(transactionContext, callback, ...args);
+  }
+
+  getTransactionContext(): TransactionContext {
+    return this._transactionContextPropagator.getTransactionContext();
```

Comment:
We should think about whether it's better to throw (and therefore default) if something goes wrong here, or just ignore the transaction context.

For example, if `getTransactionContext` is not a function, that will be an exception, which will cause a default.... or we could check that is a function, and return undefined. I think throwing/defaulting is PROBABLY better, but I'm not sure.

We could also do some validation at registration time like we do for the logger.

Reply:
fixed